tests/licensedcode/data/plugin_license/license-expression/scan.expected.json

```diff
@@ -1,78 +1,62 @@
-        "--strip-root": true
-      },
-      "notice": "Generated with ScanCode and provided on an \"AS IS\" BASIS, WITHOUT WARRANTIES\nOR CONDITIONS OF ANY KIND, either express or implied. No content created from\nScanCode should be considered or used as legal advice. Consult an Attorney\nfor any legal advice.\nScanCode is a free software code scanning tool from nexB Inc. and others.\nVisit https://github.com/nexB/scancode-toolkit/ for support and download.",
-      "output_format_version": "2.0.0",
-      "message": null,
-      "errors": [],
-      "warnings": [],
-      "extra_data": {
-        "system_environment": {
-          "operating_system": "linux",
-          "cpu_architecture": "64",
-          "platform": "Linux-5.4.0-109-generic-x86_64-with-Ubuntu-18.04-bionic",
-          "platform_version": "#123~18.04.1-Ubuntu SMP Fri Apr 8 09:48:52 UTC 2022",
-          "python_version": "3.6.9 (default, Mar 15 2022, 13:55:28) \n[GCC 8.4.0]"
-        },
-        "spdx_license_list_version": "3.16",
-        "files_count": 2
-      }
-    }
-  ],
   "files": [
     {
       "path": "apache-1.0.txt",
       "type": "file",
       "licenses": [
         {
-          "key": "apache-1.0",
-          "score": 100.0,
-          "name": "Apache License 1.0",
-          "short_name": "Apache 1.0",
-          "category": "Permissive",
-          "is_exception": false,
-          "is_unknown": false,
-          "owner": "Apache Software Foundation",
-          "homepage_url": "http://www.apache.org/licenses/",
-          "text_url": "http://www.apache.org/licenses/LICENSE-1.0",
-          "reference_url": "https://scancode-licensedb.aboutcode.org/apache-1.0",
-          "scancode_text_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/apache-1.0.LICENSE",
-          "scancode_data_url": "https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses/apache-1.0.yml",
-          "spdx_license_key": "Apache-1.0",
-          "spdx_url": "https://spdx.org/licenses/Apache-1.0",
-          "start_line": 4,
-          "end_line": 54,
-          "matched_rule": {
-            "identifier": "apache-1.0.LICENSE",
-            "license_expression": "apache-1.0",
-            "licenses": [
-              "apache-1.0"
-            ],
-            "referenced_filenames": [],
-            "is_license_text": true,
-            "is_license_notice": false,
-            "is_license_reference": false,
-            "is_license_tag": false,
-            "is_license_intro": false,
-            "has_unknown": false,
-            "matcher": "2-aho",
-            "rule_length": 368,
-            "matched_length": 368,
-            "match_coverage": 100.0,
-            "rule_relevance": 100
-          }
+          "license_expression": "apache-1.0",
+          "spdx_license_expression": "None",
+          "combination_reasons": [],
+          "matches": [
+            {
+              "key": "apache-1.0",
```

Comment:
A match is surely not for a key but for an expression (that's a problem in the current licenses output) also we do not want to repeat forever details of licenses which should be pushed out in the --license-reference . Also since a match is for a single rule, this would be best to streamline this rather than having a nested matched_rule. Along these lines, reporting matched_rule reference details only once could help too may be with a new --license-rule-reference that would report the details once

Reply:
Yes, this is now pushed, though it is WIP.